doctr/utils/geometry.py

```diff
@@ -230,9 +230,12 @@ def estimate_page_angle(polys: np.ndarray) -> float:
     """Takes a batch of rotated previously ORIENTED polys (N, 4, 2) (rectified by the classifier) and return the
     estimated angle ccw in degrees
     """
+    x1 = (polys[:, 0, 0] + polys[:, 3, 0]) / 2
+    y1 = (polys[:, 0, 1] + polys[:, 3, 1]) / 2
+    x2 = (polys[:, 1, 0] + polys[:, 2, 0]) / 2
+    y2 = (polys[:, 1, 1] + polys[:, 2, 1]) / 2
     return np.median(np.arctan(
-        (polys[:, 0, 1] - polys[:, 1, 1]) /  # Y axis from top to bottom!
-        (polys[:, 1, 0] - polys[:, 0, 0])
+        (y1 - y2) / (x2 - x1)  # Y axis from top to bottom!
```

Comment:
Since we divide them, the `/2` isn't necessary

I'm not entirely sure, but shouldn't be `np.arctan((polys[:, 3, 1] - polys[:, 0, 1]) / (polys[:, 1, 0] - polys[:, 0, 0]))`?
Introducing the average by summing won't help (on the contrary) in my opinion :thinking:

Reply:
the average is to find a middle ground when the polygon is not rectangular (or at least when the 2 sides on top and bottom in the reading direction are not parallel), we need to take the mean angle of those 2 vectors instead of choosing arbitrarily one of the sides.
Here, x1 is the "mean x left", x2 the "mean x right" with respect to the reading direction (ordered polygon) and y1 is the "mean y left", y2 the "mean y right" (always if you consider the crop to be in the reading direction), and indeed the / 2 is not necessary